src/main/java/org/tndata/android/compass/service/GcmIntentService.java

```diff
@@ -106,12 +111,12 @@ private void sendNotification(String id, String msg, String title, String object
         Context ctx = getApplicationContext();
         NotificationManager mNotificationManager = (NotificationManager)ctx.getSystemService(Context.NOTIFICATION_SERVICE);
 
-        int notificationId = (int)(Math.random()*1000000);
         if (object_type.equals(Constants.ACTION_TYPE)){
-            try {
+            try{
                 Intent intent = new Intent(getApplicationContext(), ActionActivity.class);
                 intent.putExtra(ActionActivity.ACTION_ID_KEY, Integer.valueOf(object_id));
-                //intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+
+                int notificationId = Integer.valueOf(object_id);
```

Comment:
It is (I think), and it won't, that particular statement is enclosed in an action_type notification conditional.